common/thorhelper/thorread.cpp

```diff
@@ -1272,6 +1274,205 @@ class AlternativeDiskRowReader : public CInterfaceOf<IDiskRowReader>
+    virtual unsigned numStrands() const override { return 1; };
+    virtual unsigned querySlave() const override { return curWorker; };
+    virtual unsigned queryStrand() const override { return 0; }; // 0 based 0..numStrands-1
+protected:
+    unsigned workers;
+    unsigned curWorker;
+    bool local;
+};
+
+/*
+ * Base class for reading a Parquet local file
+ */
+class ParquetDiskRowReader : public CInterfaceOf<IDiskRowStream>, implements IDiskRowReader
+{
+public:
+    ParquetDiskRowReader(IDiskReadMapping * _mapping);
+    ~ParquetDiskRowReader();
+    IMPLEMENT_IINTERFACE_USING(CInterfaceOf<IDiskRowStream>)
+
+    virtual IDiskRowStream * queryAllocatedRowStream(IEngineRowAllocator * _outputAllocator) override;
+
+    virtual const void * nextRow() override;
+    virtual const void * nextRow(size32_t & resultSize) override;
+    virtual const void * nextRow(MemoryBufferBuilder & builder) override;
+    virtual bool getCursor(MemoryBuffer & cursor) override;
+    virtual void setCursor(MemoryBuffer & cursor) override;
+    virtual void stop() override;
+
+    virtual void clearInput() override;
+    virtual bool matches(const char * _format, bool _streamRemote, IDiskReadMapping * _mapping) override;
+
+// IDiskRowReader
+    virtual bool setInputFile(const char * localFilename, const char * logicalFilename, unsigned partNumber, offset_t baseOffset, const IPropertyTree * inputOptions, const FieldFilterArray & expectedFilter) override;
+    virtual bool setInputFile(const RemoteFilename & filename, const char * logicalFilename, unsigned partNumber, offset_t baseOffset, const IPropertyTree * inputOptions, const FieldFilterArray & expectedFilter) override;
+    virtual bool setInputFile(const CLogicalFileSlice & slice, const FieldFilterArray & expectedFilter, unsigned copy) override;
+
+private:
+    template <class PROCESS>
+    inline const void * inlineNextRow(PROCESS processor) __attribute__((always_inline));
+
+protected:
+    IDiskReadMapping * mapping = nullptr;
+    Owned<IEngineRowAllocator> outputAllocator;
+
+    parquetembed::ParquetReader * parquetFileReader = nullptr;
+    CParquetActivityContext * parquetActivityCtx = nullptr;
+    StringAttr format;
+    RecordTranslationMode translationMode;
+    bool eogPending = false;
+};
+
+ParquetDiskRowReader::ParquetDiskRowReader(IDiskReadMapping * _mapping)
+ : mapping(_mapping), parquetActivityCtx(new CParquetActivityContext(true, 1, 0))
+{
+    translationMode = mapping->queryTranslationMode();
+}
+
+ParquetDiskRowReader::~ParquetDiskRowReader()
+{
+    if (parquetFileReader)
+    {
+        delete parquetFileReader;
+    }
+
+    if (parquetActivityCtx)
+    {
+        delete parquetActivityCtx;
+    }
+}
+
+IDiskRowStream * ParquetDiskRowReader::queryAllocatedRowStream(IEngineRowAllocator * _outputAllocator)
+{
+    outputAllocator.set(_outputAllocator);
+    return this;
+}
+
+const void *ParquetDiskRowReader::nextRow()
```

Comment:
Minor: Space needed after asterisk (here and in many methods below).

Reply:
Fixed.